src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,54 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] data = parseData(getData(fromFileName));
+        String report = createReport(data);
+        writeReport(toFileName, report);
+    }
+
+    private List<String> getData(String fileName) {
+        List<String> allLines;
+        try {
+            allLines = Files.readAllLines(Path.of(fileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fileName, e);
+        }
+        return allLines;
+    }
+
+    private void writeReport(String fileName, String report) {
+        try {
+            Files.createFile(Path.of(fileName));
+            Files.write(Path.of(fileName), report.getBytes());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file " + fileName, e);
+        }
+    }
+
+    private String createReport(int[] data) {
+        int result = data[0] - data[1];
+        return "supply," + data[0] + System.lineSeparator()
+                + "buy," + data[1] + System.lineSeparator()
+                + "result," + result;
+    }
 
+    private int[] parseData(List<String> data) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        for (String line : data) {
+            String[] typeAndAmount = line.split(",");
+            if (typeAndAmount[0].equals("supply")) {
```

Comment:
see checklist - If you have strange strings or numbers in the code, it’s better to declare them as constants. The name of the constant should display this object’s purpose.

Reply:
Done.